### ISSUE TITLE
MariaDb (Mysql) int column is mapped to unit32 data type

### ISSUE THREAD

**Issue by xuanduc987** [fsprojects/SQLProvider#437]
### Description

MariaDb int column is mapped to unit32 data type, causing overflow exception in runtime

![image](https://user-images.githubusercontent.com/1186411/28205268-40ea42ca-68bc-11e7-95a7-141f1cc4a24a.png)

### Repro steps

1. Create DB with int column in MariaDb

2. Create a record with negative value

3. Access that record via SQLProvider Mysql

### Expected behavior

Get value normaly

### Actual behavior

```
System.OverflowException: Value was either too large or too small for a UInt32.
   at System.Convert.ToUInt32(Int32 value)
   at System.Int32.System.IConvertible.ToUInt32(IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType)
   at FSharp.Data.Sql.Common.SqlEntity.GetColumn[T](String key)
   at <StartupCode$FSI_0004>.$FSI_0004.main@()
Stopped due to error
```

### Known workarounds

None yet

### Related information 

* Used database: MariaDb 10.2.6
* Operating system Windows 10
* Branch
* .NET Runtime, CoreCLR or Mono Version: 4.5.2
* Performance information, links to performance testing scripts

**Comment by Thorium** [comment]
I totally agree, MySql `INT` should be mapped to .NET `int` and MySQL `INT UNSIGNED` should be mapped to .NET `UInt32`. Just to note, change to this is a major breaking change for existing users.

I think the current mapping comes from MySQL.Data library reference directly and is not overridden in our code...?

**Comment by V1nzi** [comment]
I have a similar issue where all my primary keys (which are all int) are mapped to uint32. This is not a problem until I try to map the database entities (including the primary key) to a domain model using the .MapTo-method as described in [this part](http://fsprojects.github.io/SQLProvider/core/mappers.html) of the documentation.
It seems like the column is actually mapped to int32 at runtime, but uint32 at compile time, because I get a ConvertException saying that int32 cannot be converted to uint32 when trying to run my project.

One workaround would probably be to just create a custom map function and convert the datatypes as needed.

### Related information

- database: MariaDb 10.2.7
- OS: Arch Linux
- .NET runtime: Mono 5.0.0
- target framework: .NET 4.6.1
- using Monodevelop as IDE

**Comment by Thorium** [comment]
Ok, I think I found the problem. Using Mysql.Data.dll 

```fsharp
#r "Mysql.Data.dll"
#r "System.Configuration.dll"
#r "System.Transactions.dll"

open System
open MySql.Data
open MySql.Data.MySqlClient
open System.Transactions

// use your database here
let connectionString = "server = localhost; database = test; uid = root;pwd=root;"

let con = new MySqlConnection(connectionString)
con.Open()
let rows = con.GetSchema("DataTypes").Rows
for r in rows do
    printfn "%O -> %O" (r.ItemArray.GetValue(0)) (r.ItemArray.GetValue(5))
con.Close()
```

Gives with MariaDB the following mapping:
```
BIT -> System.UInt64
BLOB -> System.Byte[]
TINYBLOB -> System.Byte[]
MEDIUMBLOB -> System.Byte[]
LONGBLOB -> System.Byte[]
BINARY -> System.Byte[]
VARBINARY -> System.Byte[]
DATE -> System.DateTime
DATETIME -> System.DateTime
TIMESTAMP -> System.DateTime
TIME -> System.TimeSpan
CHAR -> System.String
NCHAR -> System.String
VARCHAR -> System.String
NVARCHAR -> System.String
SET -> System.String
ENUM -> System.String
TINYTEXT -> System.String
TEXT -> System.String
MEDIUMTEXT -> System.String
LONGTEXT -> System.String
DOUBLE -> System.Double
FLOAT -> System.Single
TINYINT -> System.SByte
SMALLINT -> System.Int16
INT -> System.Int32
YEAR -> System.Int32
MEDIUMINT -> System.Int32
BIGINT -> System.Int64
DECIMAL -> System.Decimal
TINY INT -> System.Byte
SMALLINT -> System.UInt16
MEDIUMINT -> System.UInt32
INT -> System.UInt32
BIGINT -> System.UInt64
```
If we run this through, we will get the second last: `INT -> System.UInt32`
ItemArray has also some more complete names. I'll see if I can fix it...

**Comment by xuanduc987** [comment]
One naive way to fix it would be instead of `data_type` here, we use `column_type`
https://github.com/fsprojects/SQLProvider/blob/09161c203b6abe73a74c469e3ffb148a9ff35da6/src/SQLProvider/Providers.MySql.fs#L515-L524
So instead of `int`, we got `int(11)` or `int(11) unsigned`, then remove `(_)` part.

And here
https://github.com/fsprojects/SQLProvider/blob/09161c203b6abe73a74c469e3ffb148a9ff35da6/src/SQLProvider/Providers.MySql.fs#L139-L144

We check `r.["IsUnsigned"]` and add "UNSIGNED" to TypeName when necessary.

Only problems is normal tinyint typename is `TINYINT` but unsigned tinyint typename is `TINY INT`

I'm not familiar with the code base, just comment on what I found.

Hope this helps
